distribution/ddf-common/src/main/resources/bin/ddfsolr

```diff
@@ -0,0 +1,121 @@
+
+while getopts ":u" o; do
+    case "${o}" in
+        u )
+            SECURE=false
+            ;;
+
+        * )
+            usage
+            ;;
+    esac
+done
+shift $((OPTIND-1))
+
+# -----------------------------------------------------------------------------
+if [ ! $COMMAND ]; then
+    echo "Missing command" >&2
+    usage
+fi
+
+if  ! ( [ "$COMMAND" == "start" ] || [ "$COMMAND" == "stop" ] ); then
+    echo "Unknown command: $COMMAND" >&2
+    usage
+fi
+
+SCRIPTDIR=$(dirname $0)
+HOME_DIR=$(cd "${SCRIPTDIR}/.."; pwd -P)
+GET=${SCRIPTDIR}/get_property
+SOLR_EXEC=${HOME_DIR}/solr/bin/solr
+SOLR_RC=0
+
+# Read port from system.properties file
+SOLR_PORT=$($GET "solr.http.port")
+
+# Do not proceed if required properties are missing
+if [ ! $SOLR_PORT ]; then
+    echo "Cannot determine Solr port from system.properties file. Exiting."
+    exit 1
+fi
+
+
+# -----------------------------------------------------------------------------
+# Start Solr using Solr's script
+start_solr() {
+
+    DENY_REMOTE_REQUESTS=""
+    if [ $SECURE = true ]; then
+        DENY_REMOTE_REQUESTS="-Dsolr.jetty.host=127.0.0.1"
+    fi
+
+    # To deny remote requests if TLS is NOT used, then the parameter changes
+    # to "-Djetty.host=127.0.0.1"
+
+    # Execute the start script
+    # Does not check to see if Solr is running the Solr port; there is no guarantee
+    # the Solr server is for this instance of DDF. It could have been started elsewhere.
+    $SOLR_EXEC start -force -p $SOLR_PORT $DENY_REMOTE_REQUESTS
+    SOLR_RC=$?
+}
+
+# -----------------------------------------------------------------------------
+# Set special environment variables used by Solr to configure TLS
+set_solr_https_properties() {
+      # Use the same key and trust stores as the DDF uses.
+      export SOLR_SSL_KEY_STORE=$HOME_DIR/$($GET javax.net.ssl.keyStore)
+      export SOLR_SSL_KEY_STORE_PASSWORD=$($GET javax.net.ssl.keyStorePassword)
+      export SOLR_SSL_KEY_STORE_TYPE=$($GET javax.net.ssl.keyStoreType)
+      export SOLR_SSL_TRUST_STORE=$HOME_DIR/$($GET javax.net.ssl.trustStore)
+      export SOLR_SSL_TRUST_STORE_PASSWORD=$($GET javax.net.ssl.trustStorePassword)
+      export SOLR_SSL_TRUST_STORE_TYPE=$($GET javax.net.ssl.trustStoreType)
+
+      # Require two-way TLS. Change this value to false to disable client authentication.
+      export SOLR_SSL_NEED_CLIENT_AUTH=true
+
+      # Not used
+      export SOLR_SSL_WANT_CLIENT_AUTH=false
+}
+
+# -----------------------------------------------------------------------------
+# Change these values to alter the memory available to the managed Solr JVM.
+set_solr_memory() {
+    export SOLR_JAVA_MEM="-Xms512m -Xmx2g"
+}
+
+# -----------------------------------------------------------------------------
+stop_solr() {
+    $SOLR_EXEC stop -p $SOLR_PORT
+}
+
+# -----------------------------------------------------------------------------
+if [ $COMMAND = start ]; then
+    set_solr_memory
+    if [ $SECURE = true ]; then
```

Comment:
In general it is good practice to put qoutes around variable usages in bash comparisons and around any strings used in the comparison. Here it would just be `"$SECURE"`, `true` will work both quoted and unquoted in this case

Reply:
I'm a minimalist at heart. Extra characters cause me psychic pain. But in the interest of robust software, I will add the quotes.

Reply:
Quoting variables is absolutely critical in shell scripts, especially if you don't have a default value assigned. If the variable is empty and expands, you will end up with an imbalanced comparison at best. At worst, the script will behave in interesting, unexpected, ways. The most common of which, entering into conditional blocks that you did not expect possible in the given situation.